integration-tests/modules/__tests__/cart/store/cart.workflows.spec.ts

```diff
@@ -627,6 +638,337 @@ medusaIntegrationTestRunner({
+            })
+
+            await remoteLink.create([
+              {
+                [Modules.PRODUCT]: {
+                  variant_id: product.variants[0].id,
+                },
+                [Modules.PRICING]: {
+                  price_set_id: priceSet.id,
+                },
+              },
+              {
+                [Modules.SALES_CHANNEL]: {
+                  sales_channel_id: salesChannel.id,
+                },
+                [Modules.STOCK_LOCATION]: {
+                  stock_location_id: location.id,
+                },
+              },
+              {
+                [Modules.PRODUCT]: {
+                  variant_id: product.variants[0].id,
+                },
+                [Modules.INVENTORY]: {
+                  inventory_item_id: inventoryItem.id,
+                },
+              },
+            ])
+
+            setPricingContextHook = function (input) {
+              expect(input.region).toEqual(
+                expect.objectContaining({
+                  id: region.id,
+                })
+              )
+              expect(input.variantIds).toEqual([product.variants[0].id])
+              expect(input.salesChannel).toEqual(
+                expect.objectContaining({
+                  id: salesChannel.id,
+                })
+              )
+              expect(input.customerData).toEqual(
+                expect.objectContaining({
+                  email: "tony@stark.com",
+                })
+              )
+
+              return new StepResponse({
+                unit_price: 200,
+                currency_code: "inr",
+                customer_id: "1",
+                region_id: "1",
+              })
+            }
+
+            /**
+             * Tried jest, but for some reasons it is not able to provide
+             * correct arguments passed to the function
+             */
+            let pricingContext: any
+            const originalFn = pricingModule.listPriceSets.bind(pricingModule)
+            pricingModule.listPriceSets = function () {
+              pricingContext = { ...arguments[0].context }
+              return originalFn.bind(pricingModule)(...arguments)
+            }
+
+            const { result } = await createCartWorkflow(appContainer).run({
+              input: {
+                email: "tony@stark.com",
+                currency_code: "usd",
+                region_id: region.id,
+                sales_channel_id: salesChannel.id,
+                items: [
+                  {
+                    variant_id: product.variants[0].id,
+                    quantity: 1,
+                  },
+                ],
+              },
+            })
+
+            setPricingContextHook = undefined
+            pricingModule.listPriceSets = originalFn
+
+            expect(pricingContext).toEqual(
+              expect.objectContaining({
+                unit_price: 200,
+                region_id: region.id,
+                currency_code: "usd",
```

Comment:
should we also add `customer_id` on the check?

Reply:
Just realized, that check is here. https://github.com/medusajs/medusa/pull/11891/files#diff-3084247c8dbb6a88ec078934d653e7c0e243bd1f59a2d7f982e94c451f2bcffaR947

This is because I do not have the customer_id handy. Its pulled under the hood from the email